lib/Runtime/ByteCode/IWasmByteCodeWriter.h

```diff
@@ -0,0 +1,45 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft Corporation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#ifdef ENABLE_WASM
+namespace Js
+{
+    struct IWasmByteCodeWriter
+    {
+    public:
+        virtual void Create() PURE;
+        virtual void Begin(FunctionBody* functionWrite, ArenaAllocator* alloc) PURE;
+        virtual void End() PURE;
+        virtual void Reset() PURE;
+        virtual ByteCodeLabel DefineLabel() PURE;
```

Comment:
nit: we do = 0 everywhere else, so would prefer that. i had to look up the macro definition